clients/da-vinci-client/src/main/java/com/linkedin/davinci/replication/merge/RmdSerDe.java

```diff
@@ -108,10 +121,38 @@ private RecordDeserializer<GenericRecord> getRmdDeserializer(final int writerSch
     return this.deserializerCache.get(writerSchemaID, readerSchemaID);
   }
 
+  private RecordDeserializer<GenericRecord> getRmdDeserializerWithRetry(
+      int writerSchemaID,
+      int readerSchemaID,
+      int maxRetry) {
+    return RetryUtils.executeWithMaxAttempt(() -> {
+      try {
+        return this.deserializerCache.get(writerSchemaID, readerSchemaID);
+      } catch (VeniceException e) {
+        // Adjusting the error message to include the store name and schema IDs for better debugging.
+        throw new VeniceException(
+            String.format(
+                "Failed to fetch RMD deserializer for store: %s with writer schema ID: %d and reader schema ID: %d",
+                annotatedStoreSchemaCache.getStoreName(),
+                writerSchemaID,
+                readerSchemaID),
+            e);
+      }
+    },
+        maxRetry,
+        Duration.ofSeconds(DEFAULT_DELAY_TIME_BETWEEN_RETRIES),
+        Collections.singletonList(VeniceException.class));
+  }
+
   private RecordSerializer<GenericRecord> generateRmdSerializer(int valueSchemaId) {
     Schema replicationMetadataSchema = getRmdSchema(valueSchemaId);
     return fastAvroEnabled
         ? MapOrderPreservingFastSerDeFactory.getSerializer(replicationMetadataSchema)
         : MapOrderPreservingSerDeFactory.getSerializer(replicationMetadataSchema);
   }
+
+  // For testing purpose only.
+  public void setDeserializerCache(BiIntKeyCache<RecordDeserializer<GenericRecord>> deserializerCache) {
```

Comment:
nit: for unit test no need to use public modifier.

Reply:
sure, fixed.